pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -83,6 +86,46 @@ func (p *staticPolicy) Start(s state.State) error {
 	return nil
 }
 
+func (p *staticPolicy) updateMemoryToReuse(pod *v1.Pod, container *v1.Container, blocks []state.Block){
+	// This will flush reusable resources for a pod after all containers have been allocated
+	// for podUID := range p.memoryToReuse {
+	// 	if podUID != string(pod.UID) {
+	// 		delete(p.memoryToReuse, podUID)
+	// 	}
+	// }
+
+	// Check if the container is an init container.
+	// If so, add its memory and hugepages to the reusable pool for any new allocations.
+	for _, initContainer := range pod.Spec.InitContainers {
+		if container.Name == initContainer.Name {
+			if _, ok := p.memoryToReuse[string(pod.UID)]; !ok {
+				p.memoryToReuse[string(pod.UID)] = make(map[v1.ResourceName]state.Block)
+			} 
+			for _, block := range blocks {
+				if _, ok := p.memoryToReuse[string(pod.UID)][block.Type]; !ok {
+					p.memoryToReuse[string(pod.UID)][block.Type] = block
+					continue
+				}
+				reusableBlock := p.memoryToReuse[string(pod.UID)][block.Type]
+				reusableBlock.Size += block.Size
+				p.memoryToReuse[string(pod.UID)][block.Type] = reusableBlock
+			}
+			return
+		}
+	}
+
+	//Otherwise it is an app container.
+	//Remove its memory and hugepages from the reusable pool for any new allocations.
+	for _, block := range blocks {
+		if _, ok := p.memoryToReuse[string(pod.UID)]; !ok {
+			continue
+		} 
+		reusableBlock := p.memoryToReuse[string(pod.UID)][block.Type]
+		reusableBlock.Size -= block.Reused
+		p.memoryToReuse[string(pod.UID)][block.Type] = reusableBlock
```

Comment:
It lacks checking edge cases (i.e. what if `block.Reused` > `reusableBlock.Size`?), right now it will produce not nice very high value (edge value of `uint64). I added to checklist to add error handling for this func).